### PR TITLE
Fix several bugs with matching implicit relations

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

**Comment by aelred** [comment]
!rtg - not sure what happened here: https://jenkins.grakn.ai/job/unit-test-pull-request-grakn-titan/ai.grakn$titan-factory/4085/console

**Comment by aelred** [comment]
!rtg - another cassandra timeout

File: grakn-graql/src/main/java/ai/grakn/graql/internal/util/CommonUtil.java [hunk 1]
```diff
@@ -48,6 +49,18 @@
 
     private CommonUtil() {}
 
+    public static <T> T withImplicitConceptsVisible(GraknGraph graph, Supplier<T> function) {
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/util/CommonUtil.java]: Might be worth using these methods across the whole stack. I.e. put it in grakn-core and use it everywhere.

Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/util/CommonUtil.java]: That's true. Do you know a good place to put them?

I could also just move the whole of `CommonUtil` into `grakn-core`.

Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/util/CommonUtil.java]: You mean like I suggested in my comment?

Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/util/CommonUtil.java]: Oh, you're saying just move the whole class? I like that!

Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/util/CommonUtil.java]: Sorry. Should have been more clear. My bad writing never ceases to confuse :)

**Pull request merged**